pkg/ipcache/kvstore.go

```diff
@@ -60,7 +121,43 @@ func upsertToKVStore(ipKey string, ipIDPair identity.IPIdentityPair) error {
 		logfields.Modification: Upsert,
 	}).Debug("upserting IP->ID mapping to kvstore")
 
-	return kvstore.Update(ipKey, marshaledIPIDPair, true)
+	r.Lock()
+	defer r.Unlock()
+	refcnt, ok := r.keys[ipKey]
```

Comment:
No need for all this logic. `r.keys[ipKey]` will return 0 if not found. So these 6 lines can be simplified into:
```
refcnt := r.keyx[ipKey] // 0 if not found
refcnt++
```